vulkan_helpers/vulkan_application.cpp

```diff
@@ -198,7 +198,7 @@ VulkanApplication::VulkanApplication(
 
   VkBufferUsageFlags usages[3] = {
       VK_BUFFER_USAGE_TRANSFER_SRC_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
-      kAllBufferBits, kAllBufferBits};
+      kAllBufferBits, VK_BUFFER_USAGE_TRANSFER_SRC_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT | VK_BUFFER_USAGE_STORAGE_BUFFER_BIT};
```

Comment:
What was the reason for the previous structure with 2 usages as kAllBufferBits? or was that a bug?

Reply:
This was so that you could allocate "ANY" buffer in our coherent arena. But it turns out that is not always possible. 
(This is a gross oversimplification of how a normal game SHOULD do memory management, but it keeps things simple for us)